test/auth/logic/credential_test.clj

```diff
@@ -0,0 +1,38 @@
+(ns auth.logic.credential-test
+  (:require [midje.sweet :refer :all]
+            [auth.logic.credential :as logic.credential]
+            [common-labsoft.misc :as misc]
+            [common-labsoft.protocols.crypto :as protocols.crypto]))
+
+(def user-id (misc/uuid))
+(def pass-creds {:credential/type :credential.type/password
+                 :credential/email "email@test.com"
+                 :credential/encrypted-password "bcryptedpass"
+                 :credential/user-id user-id})
+
+(facts "when we create new credentials"
+  (fact "on `base-credential`"
+    (logic.credential/base-credential user-id "email@test.com" :credential.type/password)
+    => {:credential/type :credential.type/password
+        :credential/email "email@test.com"
+        :credential/user-id user-id}))
+
+(facts "utilities functions for credentials"
+  (fact "on `valid-cred-type?`"
```

Comment:
https://github.com/marick/Midje/wiki/Tabular-facts

Reply:
I know, it was just laziness 😅